packages/react-component-library/src/components/CheckboxRadioBase/CheckboxRadioBase.tsx

```diff
@@ -36,7 +37,13 @@ export const CheckboxRadioBase = React.forwardRef<
   ) => {
     const id = useExternalId(`${type}-input`, externalId)
     const localRef = useRef<HTMLInputElement>(null)
-    const [isChecked, setIsChecked] = useState(defaultChecked)
+    const [isChecked, setIsChecked] = useState<boolean>(
+      defaultChecked || checked
```

Comment:
`??` might make more sense here (i.e. still use `defaultChecked` if it's false rather than undefined):

```suggestion
      defaultChecked ?? checked
```